### ISSUE TITLE
Partial mentions should not be downcasted (not upcasted, like in the docs)

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-mention#24]
Part of the documentation:

```js
function MentionCustomization( editor ) {
	// The upcast converter will convert <a class="mention"> elements to the model 'mention' attribute.
	editor.conversion.for( 'upcast' ).elementToAttribute( {
		view: {
			name: 'a',
			key: 'data-mention',
			classes: 'mention',
			attributes: {
				href: true,
				'data-user-id': true
			}
		},
		model: {
			key: 'mention',
			value: viewItem => {
				// Optional: Do not convert partial mentions.
				if ( !isFullMention( viewItem ) ) {
					return;
				}

				// The mention feature expects that mention attribute value in the model is a plain object:
				const mentionValue = {
					// The name attribute is required by mention editing.
					name: viewItem.getAttribute( 'data-mention' ),
					// Add any other properties as required.
					link: viewItem.getAttribute( 'href' ),
					id: viewItem.getAttribute( 'data-user-id' )
				};

				return mentionValue;
			}
		},
		converterPriority: 'high'
	} );

	function isFullMention( viewElement ) {
		const textNode = viewElement.getChild( 0 );
		const dataMention = viewElement.getAttribute( 'data-mention' );

		// Do not parse empty mentions.
		if ( !textNode || !textNode.is( 'text' ) ) {
			return false;
		}

		const mentionString = textNode.data;

		// Assume that mention is set as marker + mention name.
		const name = mentionString.slice( 1 );

		// Do not upcast partial mentions - might come from copy-paste of partially selected mention.
		return name == dataMention;
	}
```

IMO, partial mentions should never end up in the data (any sort of data – also the data clipboard). No one should worry about that except us in the mention feature's code.

**Comment by Reinmar** [comment]
I'm removing this whole bit from the docs.

**Comment by jodator** [comment]
This might be obsolete after https://github.com/ckeditor/ckeditor5-mention/pull/17. To be checked.

**Comment by jodator** [comment]
So the Mention allows to downcast part of mention:
This is copied to clipboard:
```html
<meta http-equiv="content-type" content="text/html; charset=utf-8">Hello <span class="mention" data-mention="@Ted">@Te</span>
```

**Comment by Reinmar** [comment]
What do you mean? Do you describe an expected or actual behaviour before or after that PR? :D

**Comment by jodator** [comment]
It is current behavior (copy partial mention) and is fixed on PR (will not copy partial mention).

I've added this comment because in my previous one I thought that #17 might fix that issue. I've checked this and it is still present (you can copy partial mention on current master).

**Comment by jodator** [comment]
One note in some earlier works we concluded that partial mention from data will be converter as-is. So it might be a case for a partial mention. We did this because we're not able to verify if the mention is partial when we upcast it to the model.

We could do this asynchronously using feed callback and verify each mention if it is rendered as a whole mention. I don't recall who was proposing this (maybe it was @scofalik or @pjasiun) but we could do conversion asynchronously (feed might be asynchronous).

Just to be clear :)
